src/loaders.ts

```diff
@@ -77,14 +78,22 @@ const emptyContractResult: ContractResult = {
 }
 
 export interface ABILoader {
+    readonly name: string;
+
     loadABI(address: string): Promise<any[]>;
     getContract(address: string): Promise<ContractResult>
 }
 
 // Load ABIs from multiple providers until a result is found.
 export class MultiABILoader implements ABILoader {
+    readonly name: string = "MultiABILoader";
+
     loaders: ABILoader[];
 
+    // Note: This callback is used to pull out which loader succeeded without modifying the return API.
+    // We can remove it once we switch to using getContract for autoload.
+    onLoad?: (loader: ABILoader) => void;
```

Comment:
I had to do some nasty things here to pull out the successful loader out of MultiABILoader.loadABI without changing the interface, but probably worth it for now. I think we can get rid of it we're able to switch to getContract later (sourcify is releasing a new API that has selective field loading, that might be worth it).